### PR TITLE
Feature/account page

### PR DESCRIPTION
**Pull request by anmol-fzr**
### What changes did you make?
Added Account Page with UI + Functionality + Translations. Delete Account button is intentionally left disabled reason being API to delete account by user isn't ready 

### Why did you make the changes?
Changes are made to fix the issue 

### Did you run tests?
Added a test named user-profile-update that ran successfully

### CODE REVIEW

**Comment by eleanorreem** [comment]
Thanks so much for this, loads of work 🎉  

I know that this PR is WIP so apologies for all the comments. I had a look through the code in combination with the current functionality that we had in the backend. These are my comments: 
- [ ] The backend for updating an email isn't implemented so for now it is best just to have the _Profile details_ form as per the designs (i.e. add the link in the issue to the _getting in touch_ text and remove the button). If you would like to contribute this work at a later date, I can tag you in the issue when it comes up and you can simply implement the work you have already done. Let me know if that works for you!
- [ ] The backend for deleting an account isn't ready yet and is going to be a bit complicated. For the sake of getting this merged soon, I would remove the _Account Actions_ section for the PR and save it for the next issue that I will write up in the near future. 
- [ ] The email subscription form works well. I am happy with it. We will need to have some error / success feedback for the user. Please see the bloom-frontend/components/forms/CreateAccessCodeForm.tsx for some input on how you might implement that. For now the success can just be "Your email subscription preferences have been updated" 
- [ ] There are some console logs and commented out code that will need tidying up. 
- [ ] As you have included more functionality in this PR than in the original ticket, you would need to add more functionality to the cypress test. In the new cypress test, you will need to check the profile details and that updating your subscription preferences works. 
- [ ] The header h1 and description isn't quite aligned with the middle of the image. You might need to add a marginTop: auto and a marginBottom: auto the top and bottom of the div that contains the header and description. I hope that makes sense. 
- [ ] There are errors when I run yarn build and yarn lint so it would be great if you could sort the single quotes. 
- [ ] When you have made all the changes, it would be awesome if you could soft reset the branch and redo the commits, excluding the feature we aren't going to include in this PR otherwise it might get confusing that we are adding feature and then taking them away. 

Sorry for the long list of comments! What you have done so far is great. Thank you for all your work. Please do keep the work that isn't going to be included in this PR and we can use it in the very near future.

File: guards/publicPageDataWrapper.tsx [hunk 1]
```diff
@@ -75,12 +73,7 @@ export function PublicPageDataWrapper({ children }: { children: JSX.Element }) {
           logEvent(GET_AUTH_USER_ERROR, { message: getErrorMessage(userResponse.error) });
         }
 
-        signOut(auth);
-        await dispatch(clearPartnerAccessesSlice());
-        await dispatch(clearPartnerAdminSlice());
-        await dispatch(clearCoursesSlice());
-        await dispatch(clearUserSlice());
-        await dispatch(api.util.resetApiState());
+        await onLogout()
```
Comment (eleanorreem) [inline guards/publicPageDataWrapper.tsx:76]: I like this refactor

File: components/forms/LoginForm.tsx [hunk 2]
```diff
@@ -30,17 +29,16 @@ const containerStyle = {
...
   const [formError, setFormError] = useState<
     | string
-    | React.ReactNodeArray
+    | React.ReactNode
     | React.ReactElement<any, string | React.JSXElementConstructor<any>>
   >();
   const [emailInput, setEmailInput] = useState<string>('');
   const [passwordInput, setPasswordInput] = useState<string>('');
-  const [getUser, { isLoading: getUserIsLoading }] = useGetUserMutation();
+  const [getUser] = useGetUserMutation();
```
Comment (eleanorreem) [inline components/forms/LoginForm.tsx:41]: Just wondering about these changes - I assume this code was unused?

File: components/forms/LoginForm.tsx [hunk 3]
```diff
@@ -91,7 +89,6 @@ const LoginForm = () => {
       })
       .catch((error) => {
         const errorCode = error.code;
-        const errorMessage = error.message;
```
Comment (eleanorreem) [inline components/forms/LoginForm.tsx:92]: Same here - assume this was unused code?

File: pages/account/settings.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  const partnerAccesses = useTypedSelector((state) => state.partnerAccesses);
+  const partnerAdmin = useTypedSelector((state) => state.partnerAdmin);
+  const eventUserData = getEventUserData(userCreatedAt, partnerAccesses, partnerAdmin);
+  const [error, setError] = useState<
+    string | ReactElement<any, string | JSXElementConstructor<any>> | ReactNode | null
+  >();
+  const [updateUser, { isLoading: updateUserIsLoading }] = useUpdateUserMutation();
+
+  useEffect(() => {
+    if (userCreatedAt && userServiceEmailsPermission === true) {
```
Comment (eleanorreem) [inline pages/account/settings.tsx:44]: I am not 100% on the purpose of this use effect. Can you explain?

File: pages/account/settings.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    <Box>
+      <Head>
+        <title>{t('title')}</title>
+      </Head>
+      {error && (
+        <Container>
+          <Typography>{error}</Typography>
+        </Container>
+      )}
+      {!error && !updateUserIsLoading && (
```
Comment (eleanorreem) [inline pages/account/settings.tsx:81]: I think this wrapper around the header is causing an issue. Why have you wrapped it in an error?

File: components/account/EmailPref.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+} as const;
+
+const EmailPref = () => {
+  const [updateUser, { isLoading }] = useUpdateUserMutation()
+  const t = useTranslations('Account.accountSettings.emailPref');
+
+  const cPerms = useTypedSelector(state => state.user.contactPermission)
+  const sPerms = useTypedSelector(state => state.user.serviceEmailsPermission)
+
+  console.log({ cPerms, sPerms })
```
Comment (eleanorreem) [inline components/account/EmailPref.tsx:22]: console.logs and commented out code

File: pages/account/settings.tsx (hunk 4 above)
Comment (eleanorreem) [reply pages/account/settings.tsx:44]: I think I see where you have copied this from. This shouldn't be there, that functionality is only for the disable-service-email.tsx page.

Comment (eleanorreem) [reply pages/account/settings.tsx:44]: A better model for this page should be the `apply a code` page.

File: components/account/EmailPref.tsx [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+  const onSubmit = useCallback(async (ev: React.FormEvent<HTMLFormElement>) => {
+    const formData = new FormData(ev.currentTarget)
+    ev.preventDefault()
+
+    const contactPermission = formData.get('cPerms') === 'on'
+    const serviceEmailsPermission = formData.get('sPerms') === 'on'
+    const payload = {
+      contactPermission, serviceEmailsPermission
+    }
+    updateUser(payload)
```
Comment (eleanorreem) [inline components/account/EmailPref.tsx:39]: Add an await and ensure that we are handling the error/ success scenarios 😄

File: components/account/EmailPref.tsx [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+                }
+              />
+              <FormControlLabel
+                label={t('checkbox.emailOnCourse')}
+                control={
+                  <Checkbox
+                    name='sPerms'
+                    aria-label={t('checkbox.emailOnCourse')}
+                    defaultChecked={sPerms}
+                  // checked={emailPerms}
```
Comment (eleanorreem) [inline components/account/EmailPref.tsx:73]: commented out code ⭐

File: cypress/integration/user-profile-update.cy.tsx [hunk 9]
```diff
@@ -0,0 +1,32 @@
+describe('User should be able to update his profile', () => {
+  const publicEmail = Cypress.env("public_email")
+
```
Comment (eleanorreem) [inline cypress/integration/user-profile-update.cy.tsx:3]: See overall comment about what sort of functionality this should be testing.

File: pages/account/settings.tsx [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+    imageSrc: illustrationPerson5Yellow,
+    translatedImageAlt: t('imageAlt'),
+  };
+
+  return (
+    <Box>
+      <Head>
+        <title>{t('title')}</title>
+      </Head>
+      {error && (
```
Comment (eleanorreem) [inline pages/account/settings.tsx:76]: Not sure what this error wrapper is doing around the container?

File: components/account/AccountActions.tsx [hunk 11]
```diff
@@ -0,0 +1,49 @@
+import { Typography, CardContent, Card, Button } from '@mui/material'
```
Comment (eleanorreem) [inline components/account/AccountActions.tsx:1]: For now, lets exclude this from the PR. I can review this properly in the next iteration of this page. 🙏 Thanks so much for doing it though!

File: components/forms/ProfileDetailsForm.tsx [hunk 12]
```diff
@@ -0,0 +1,104 @@
+import { Box, TextField, Container, Typography } from '@mui/material';
+import { useTranslations } from 'next-intl';
```
Comment (eleanorreem) [inline components/forms/ProfileDetailsForm.tsx:2]: For now, lets just display the name and email and ask them to fill out their form details. Keep the code for another pull request ⭐

**Comment by anmol-fzr** [comment]
@eleanorreem I made the code changes suggested in the chat please take a look in to it.

File: pages/account/settings.tsx [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+              {t('profile.title')}
+            </Typography>
+            <Typography fontSize="1rem !important">
+              {t('profile.desc')}
+            </Typography>
+            <ProfileDetailsForm />
+          </CardContent>
+        </Card>
+
+        <AccountActions />
```
Comment (eleanorreem) [inline pages/account/settings.tsx:61]: Could you remove the Account actions card from the page and take the code from the PR. Thank you! We can add this to a PR at a later date

**Comment by eleanorreem** [comment]
Thanks for your changes! They look great there were a few more changes that were on PR. It would be great if they could be resolved. These are the ones outstanding:

- [ ] removing account actions from the card
- [ ] success/ error handling for saving profile information
- [ ] The cypress test doesn't test whether the correct email is displaying, it just tests that the box is disabled. Ideally you would be checking against the email address and also checking that when you update your email address, it works.

File: pages/account/settings.tsx (hunk 13 above)
Comment (anmol-fzr) [reply pages/account/settings.tsx]: okay I understand, i have commited proposed changes.

**Comment by eleanorreem** [comment]
@anmol-fzr, thanks for all your work so far! Just checking in here to make sure you have everything you need to finish off this ticket. Absolutely no pressure, it would just be great to get this contribution merged!

**Comment by anmol-fzr** [comment]
@eleanorreem , I made the changes to the code suggest in this review please check.

File: components/account/EmailPref.tsx (hunk 7 above)
Comment (eleanorreem) [reply components/account/EmailPref.tsx:30]: @anmol-fzr  let me know if you have any questions about this change

File: components/account/EmailPref.tsx (hunk 8 above)
Comment (eleanorreem) [reply components/account/EmailPref.tsx:64]: @anmol-fzr let me know if you have any questions about this change

File: cypress/integration/user-profile-update.cy.tsx (hunk 9 above)
Comment (eleanorreem) [reply cypress/integration/user-profile-update.cy.tsx]: @anmol-fzr this cypress test doesn't address the scenarios that was specified in the ticket. Let me know if you have any questions!

File: pages/account/settings.tsx (hunk 4 above)
Comment (eleanorreem) [reply pages/account/settings.tsx]: @anmol-fzr let me know if you have any questions about this change

File: components/forms/ProfileDetailsForm.tsx [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+
+const onEmailUpdate = (newEmail: string) => {
+  const auth = getAuth()
+  const currUser = auth.currentUser
+
+  if (currUser === null) {
+    return
+  }
+
+  verifyBeforeUpdateEmail(currUser, newEmail).then((res) => {
```
Comment (eleanorreem) [inline components/forms/ProfileDetailsForm.tsx]: On email update is no longer in use so should probably be removed.

**Comment by eleanorreem** [comment]
Hi thank you for your quick reply and all the work so far! There were a few more things that I commented on your pull request - take a look through the pull request files. There are some unresolved comments that need to be addressed before I can merge.

**Comment by anmol-fzr** [comment]
@eleanorreem I made change told last time about removing account action card to header alignment. It would be great if you review the code for merge or we resolve anything that needs to be done for this PR

**Comment by eleanorreem** [comment]
Hiya @anmol-fzr, Thanks for your response 😄 

There are these open comments on your PR. Let me know if you have any questions. I linked to the most of the comments here. 

- [ ] [Can you delete this file](https://github.com/chaynHQ/bloom-frontend/pull/900/files#r1597581731)
- [ ] [Can you add an await to ensure this function is async and ensure that there is a success and failure scenrio](https://github.com/chaynHQ/bloom-frontend/pull/900/files#r1597581163). I have signposted in the comment where to look for examples. 
- [ ] [You've left commented out code which would be great to be removed. 
](https://github.com/chaynHQ/bloom-frontend/pull/900/files#r1597581219)
- [ ] [I had a question about some changes](https://github.com/chaynHQ/bloom-frontend/pull/900/files#r1597577605)
- [ ] Please note the cypress test doesn’t cover the functionality of updating your subscription preferences, specified in the comments above.

Please check through the open comments and make sure they are all resolved.

Thanks in advance for everything !

**Comment by anmol-fzr** [comment]
@eleanorreem  i made suggested changes, please check

**Comment by anmol-fzr** [comment]
hope we can merge this now
